File: C#/TopflytechCodec/TopflytechCodec/Entities/AccelerationDriverBehaviorMessage.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace TopflytechCodec.Entities
{
    /// <summary>
    /// Driver Behavior Via Acceleration (AST Command Control, Default Disable the feature).Protocol number is 25 25 07.
    /// </summary>
    public class AccelerationDriverBehaviorMessage : Message
    {
        protected AccelerationData accelerationData;
        public TopflytechCodec.Entities.AccelerationData AccelerationData
        {
            get { return accelerationData; }
            set { accelerationData = value; }
        }
        private int behaviorType;
        public int BehaviorType
        {
            get { return behaviorType; }
            set { behaviorType = value; }
        }
        
        /// <summary>
        /// The constant BEHAVIOR_TURN_AND_BRAKE.
        /// </summary>
        public static int BEHAVIOR_TURN_AND_BRAKE = 0;

        /// <summary>
        ///  The constant BEHAVIOR_ACCELERATE.
        /// </summary>
        public static int BEHAVIOR_ACCELERATE = 1;

    }
}

File: Java/topflytechCodec/src/com/topflytech/codec/entities/AccelerationDriverBehaviorMessage.java
package com.topflytech.codec.entities;

/**
 * .Driver Behavior Via Acceleration (AST Command Control, Default Disable the feature).Protocol number is 25 25 06.
 */
public class AccelerationDriverBehaviorMessage extends Message{


    /**
     * Gets acceleration data.
     *
     * @return the acceleration data
     */
    public AccelerationData getAccelerationData() {
        return accelerationData;
    }


    /**
     * Sets acceleration data.
     *
     * @param acceleration the acceleration
     */
    public void setAccelerationData(AccelerationData acceleration) {
        this.accelerationData = acceleration;
    }



    /**
     * The Acceleration data.
     */
    protected AccelerationData accelerationData;

    /**
     * Gets behavior type.
     *
     * @return the behavior type
     */
    public int getBehaviorType() {
        return behaviorType;
    }

    /**
     * Sets behavior type.
     *
     * @param behaviorType the behavior type
     */
    public void setBehaviorType(int behaviorType) {
        this.behaviorType = behaviorType;
    }




    private int behaviorType;
    /**
     * The constant BEHAVIOR_TURN_AND_BRAKE.
     */
    public final static int BEHAVIOR_TURN_AND_BRAKE = 0;
    /**
     * The constant BEHAVIOR_ACCELERATE.
     */
    public final static int BEHAVIOR_ACCELERATE = 1;




}
